c7n/resources/vpc.py

```diff
@@ -1856,6 +1856,27 @@ class resource_type(query.TypeInfo):
         id_prefix = "igw-"
 
 
+@InternetGateway.action_registry.register('delete')
+class DeleteInternetGateway(BaseAction):
+
+    schema = type_schema('delete')
+    permissions = ('ec2:DeleteInternetGateway',)
+
+    def process(self, resources):
+
+        client = local_session(self.manager.session_factory).client('ec2')
+        for r in resources:
+            try:
+                client.delete_internet_gateway(InternetGatewayId=r['InternetGatewayId'])
+            except (client.exceptions.NoSuchResource):
+                continue
+            except Exception as e:
```

Comment:
You could probably omit this second except since you end up raising it anyway, unless you think that knowing the InternetGatewayId will provide some details that would help with diagnosing the error

Reply:
I think I like leaving it there, as it will also catch any unforeseen errors that are not a client.exceptions.NoSuchResource exception.

Reply:
Nevermind, the ec2 client actually has no built in exceptions at all, so I refactored the code to catch a client error where the internet gateway ID doesn't exist